5_EDA/src/select_features.R

```diff
@@ -71,5 +69,53 @@ drop_high_corr_ACCTOT <- function(features, threshold_corr, drop_var){
   #cor_mat <- abs(cor(features[,-c(1,2)] %>% select(-contains('_PPT_'), 
   #-contains('_TAV_'), -contains('WB5100'))))
   
+  
+  #Automatic removal of all other highly correlated features
+  #Correlation matrix for only static features
+  high_corr_features <- get_high_corr_features(features %>% 
+                                                 select(-COMID, -GAGES_ID), 
+                                               cor_method, threshold_corr)
+  while (nrow(high_corr_features) > 0){
+    #randomly select an attribute and remove all correlated attributes
+    tmp_col <- sample(unique(rownames(high_corr_features)), size = 1)
+    tmp_rm <- features %>% 
+      select(-COMID, -GAGES_ID) %>%
+      .[, high_corr_features[rownames(high_corr_features) == tmp_col, 2] %>%
+                             as.numeric()] %>%
+      colnames()
+    
+    features <- features %>% select(-{{tmp_rm}})
+    #Correlation matrix for features without the remove_vars
+    high_corr_features <- get_high_corr_features(features %>% 
+                                                   select(-COMID, -GAGES_ID), 
+                                                 cor_method, threshold_corr)
+  }
+  
   return(features)
 }
+
+
+get_high_corr_features <- function(features, cor_method, threshold_corr){
+  #' 
+  #' @description Function to convert a string from #days #months, #weeks, #years to 
+  #' equivalent number of days 
```

Comment:
I'm not concerned from a method perspective, but that is a good point that we'll not have exact reproducibility compared to someone else running our pipeline. `targets` does save the random seed used to build each target in the meta file, so someone could get our exact results if they really wanted to. I'm happy to change the variable selected to be the 1st element in `high_corr_features` every time instead.

Regardless, it will be important for us to explain in the paper that the features we have may represent more than just their named feature due to correlation. For the most important features, we can check their correlation matrix. I think I'll file an issue for that to remind us to do it.

Reply:
Okay, I fixed this and updated the targets. Let me know if it's good to merge @cstillwellusgs

Reply:
go for it!